### DISCUSSION TITLE
Could you please tell me how to implement automatic upgrades for the software you develop? Any suggestions?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#4314] [Q&A]
## With so many software, it is rare to see a module that specifically provides software upgrades.

**Comment by andydotxyz** [comment]
The easiest way with Fyne is to use https://github.com/fynelabs/fyneselfupdate.
Fyne Labs also has a server to provide the infrastructure automatically for you - https://fynelabs.com/geoffrey
